optuna/samplers/_tpe/sampler.py

```diff
@@ -470,20 +470,18 @@ def _gmm_log_pdf(
     def _sample_from_categorical_dist(self, probabilities, size):
         # type: (np.ndarray, Tuple[int]) -> np.ndarray
 
-        if probabilities.size == 1 and isinstance(probabilities[0], np.ndarray):
-            probabilities = probabilities[0]
-        probabilities = np.asarray(probabilities)
-
         if size == (0,):
             return np.asarray([], dtype=float)
         assert len(size)
+
+        if probabilities.size == 1 and isinstance(probabilities[0], np.ndarray):
+            probabilities = probabilities[0]
         assert probabilities.ndim == 1
 
-        n_draws = int(np.prod(size))
-        sample = self._rng.multinomial(n=1, pvals=probabilities, size=int(n_draws))
-        assert sample.shape == size + (probabilities.size,)
-        return_val = np.dot(sample, np.arange(probabilities.size))
-        return_val.shape = size
+        n_draws = np.prod(size).item()
```

Comment:
Does this line convert `float` to `int`? Why do you add the `item()` method?

Reply:
Why "convert `float` to `int`"?

The return type of `np.prod` is `np.int64` as `size` is expected to be a tuple of integers. Thus I think `item` is enough. Furthermore, previously we applied `int()` to the result of `np.prod(size)` twice.

Reply:
Thanks for the reply. I understood. It actually converts `np.int64` to `int`.